### DISCUSSION TITLE
Get Current Window URL Query Params

### DISCUSSION THREAD

**Discussion by luisdgeneMT** [Budibase/budibase#11425] [Help]
Hello,

I am attempting to get an auth token from a callback URI pointing to my app.
The current callback URI looks like this once my user authorizes in the external app:

`https://myapp.budibase.app/app/appname?code=CODE#/appPage`

In other instances I would have used something like
`return window.location.href`
and parsed CODE, since I am aware budibase uses path parameters and not query parameters when parsing the URL.query using /:query notation.

I cannot use the OpenID auth option for this specific scenario.

If it helps, I am trying to connect this way:
https://developer.xero.com/documentation/guides/oauth2/auth-flow

**Comment by ConorWebb96** [comment]
Hey @luisdgeneMT,

You can also use query parameters and output the query. Heres a quick demo below showing how this can work;

**Screen structure**;
<img width="262" alt="Screenshot 2023-08-03 at 08 20 12" src="https://github.com/Budibase/budibase/assets/126772285/27aa5d2b-6cc0-4fc2-86a5-3e4a02ddad5d">

**Home page with button navigate to action**;
<img width="1295" alt="Screenshot 2023-08-03 at 08 20 01" src="https://github.com/Budibase/budibase/assets/126772285/c1569a4e-e881-4b2f-b005-9c57995aea7b">

**Query page setup**
<img width="1293" alt="Screenshot 2023-08-03 at 08 20 22" src="https://github.com/Budibase/budibase/assets/126772285/2d3ff5c4-7f6a-4f4b-93b4-9a955e231f66">

**Output of working version**
![415de83839f4419eaa59b99f3766cfbe-00001](https://github.com/Budibase/budibase/assets/126772285/0019df68-d4d7-4472-a9f9-a7a0bb607a1d)

Hopefully this helps.

**Reply by luisdgeneMT** [reply]
Hi @ConorWebb96,

So unfortunately this is not a functional workaround due to the following:

As configured, my URI callback looks something like this (I will use your example naming the page "query"):
`https://myapp.budibase.app/app/appname/query`

When the page loads on budibase including the query parameters, it ends up looking like this:
`https://myapp.budibase.app/app/appname/query?code=123#/query`

To my understanding, your workaround would only work if it looked something like this:
`https://myapp.budibase.app/app/appname#/query?code=123`

However I have no way of getting the redirect URI to be sent this way from the Xero auth screen - my current use case.
This is why I need the entire URL, so I can parse manually, because **query params only work if passed internally** - at least that's the behavior I am seeing thus far.

Any suggestions? Thanks in advance.

**Reply by A3DataSolutions** [reply]
> Hi @ConorWebb96,
> 
> So unfortunately this is not a functional workaround due to the following:
> 
> As configured, my URI callback looks something like this (I will use your example naming the page "query"): `https://myapp.budibase.app/app/appname/query`
> 
> When the page loads on budibase including the query parameters, it ends up looking like this: `https://myapp.budibase.app/app/appname/query?code=123#/query`
> 
> To my understanding, your workaround would only work if it looked something like this: `https://myapp.budibase.app/app/appname#/query?code=123`
> 
> However I have no way of getting the redirect URI to be sent this way from the Xero auth screen - my current use case. This is why I need the entire URL, so I can parse manually, because **query params only work if passed internally** - at least that's the behavior I am seeing thus far.
> 
> Any suggestions? Thanks in advance.

Hi I am actually trying to do something very similar with Xero too!

Did you have any luck with this?

**Comment by freekvdp** [comment]
@luisdgeneMT I'd love to know if you managed to make this work as I want to try to achieve something similar
